src/app/services/User.service.ts

```diff
@@ -85,4 +84,20 @@ export class UserService {
 
     await this.repository.update(user);
   }
+
+  async updatePassword(id: string, values: changePasswordDTO) {
+    const user = await this.repository.getById(id);
+
+    if (!user) {
+      throw new NoSuchElementException('user not found');
+    }
+
+    if (values.password !== values.confirmPassword) {
+      throw new Error('Passwords do not match');
```

Comment:
I'd move this directly to the validation schema, request with non matching passwords should not even pass validation